File: prog1HMK.c
//Haruka Kido
//haruka.kido@ndus.edu
//CSci 265. Program 1, C

#include <stdio.h>
#include <string.h>

int main() 
{
    //Part 1: Asks user to input integer value as number of seconds and converts input to hours, minutes, and seconds as output
    printf("Part 1 – Convert from seconds");
    printf("\n");

	int input, hours, mins, secs;
	printf("Seconds: ");
	scanf("%d", &input);
	
	hours = (input/3600); 
	mins = (input - (3600*hours))/60;
	secs = (input - (3600*hours) - (60*mins));
	
	printf("This is %d hours, %d minutes and %d seconds.",hours,mins,secs); //displays output

    //divider print statement
    printf("\n");
    printf("\n");
    
    
    //Part 2: Asks user to input 4 integer values as numbers of quarters, dimes, nickels, and pennies and converts input to dollars and cents as output
    printf("Part 2 – Convert to dollars");
    printf("\n");

    int quarters, dimes, nickels, pennies;

    printf("Quarters: ");
    scanf("%d", &quarters);
    printf("Dimes: ");
    scanf("%d", &dimes);
    printf("Nickels: ");
    scanf("%d", &nickels);
    printf("Pennies: ");
    scanf("%d", &pennies);
    
    int dollars = 0;
    int cents = 0;

    //Equivalencies in cents
    cents += quarters * 25;
    cents += dimes * 10;
    cents += nickels * 5;
    cents += pennies;
    
    //Equivalencies in dollars and cents
    dollars = cents/100;
    cents = cents%100;

    printf("This is equal to $%d.%d", dollars, cents); //displays output

    //divider print statement
    printf("\n\n");
    
    
    //Part 3: Asks user to input first name, last name, address, city, state, and zip code (all strings) and outputs formatted inputs in 2 ways
    printf("Part 3 - Work with output");
    printf("\n");
    
    //array of chars
    char firstname[20], lastname[20], address[40], city[20], state[20], zipcode[20];
    char c;
    while((c = getchar()) != '\n' && c != EOF);
    
    //user inputs
    printf("First Name: ");
    fgets(firstname, 19, stdin);
    firstname[strcspn(firstname, "[\r\n]")] = 0;
    
    printf("Last Name: ");
    fgets(lastname, 19, stdin);
    lastname[strcspn(lastname, "[\r\n]")] = 0;
    
    printf("Address: ");
    fgets(address, 39, stdin);
    address[strcspn(address, "[\r\n]")] = 0;
    
    printf("City: ");
    fgets(city, 19, stdin);
    city[strcspn(city, "[\r\n]")] = 0;
    
    printf("State: ");
    fgets(state, 19, stdin);
    state[strcspn(state, "[\r\n]")] = 0;
    
    printf("Zip Code: ");
    fgets(zipcode, 19, stdin);
    zipcode[strcspn(zipcode, "[\r\n]")] = 0;
    
    //divider print statement
    printf("\n");
    
    //format output 1: using exactly 1 print statement
    printf("%s %s\n%s \n%s, %s  %s\n", firstname, lastname, address, city, state, zipcode);
    
    //divider print statement
    printf("\n");
    
    //format output 2: using exactly 6 print statements
    printf("%s", firstname);
    printf(" %s\n", lastname);
    printf("%s", address);
    printf("\n%s, ", city);
    printf("%s ", state);
    printf("%s\n", zipcode);
    
    return 0;
}

File: prog1HMK.py
#Haruka Kido
#haruka.kido@ndus.edu
#CSci 265. Program 1, Python

def p1(input):
#Part 1: Asks user to input integer value as number of seconds and converts input to hours, minutes, and seconds as output
    print("Part 1 - Convert from seconds")
    input = int(input("Seconds: "))
    hour = input // 3600
    input %= 3600
    minutes = input // 60
    input %= 60
    seconds = input
    print("This is " + str(hour) + " hours, " + str(minutes) + " minutes " + str(seconds) + " seconds.") #displays output


def p2():
#Part 2: Asks user to input 4 integer values as numbers of quarters, dimes, nickels, and pennies and converts input to dollars and cents as output
    print("Part 2 - Convert to dollars")
    def Input ():
        quarters = input("Quarters: ")
        dimes = input("Dimes: ")
        nickels = input("Nickels: ")
        pennies = input("Pennies: ")
        return ([quarters,dimes,nickels,pennies])

    def CalcCurrencyAmt(c):
        quarters = c[0]
        dimes = c[1]
        nickels = c[2]
        pennies = c[3]
    
        total = 25*int(quarters) + 10*int(dimes) + 5*int(nickels) + int(pennies)
        currAmt = float(total)/100
        return([total,currAmt])

    c = Input()

    conversion = CalcCurrencyAmt(c)
    print("This is equal to $" + str(conversion[1])) #displays output


def p3():
#Part 3: Asks user to input first name, last name, address, city, state, and zip code (all strings) and outputs formatted inputs in 2 ways
    print("Part 3 - Work with output")
    first_name = input("First Name: ") #string first name
    last_name = input("Last Name: ") #string last name
    address = input("Address: ") #string address
    city = input("City: ") #string city
    state = input("State: ") #string state abbreviation
    zip_code = input("Zip Code: ") #string zip code

    #divider print statement
    print("\n")

    #format output 1: using exactly 1 print statement
    print(str(first_name) + " " + str(last_name) + " \n" + str(address) + " \n" + str(city) + ", " + str(state.upper()) + "  " + str(zip_code))

    #divider print statement
    print("\n")

    #format output 2: using exactly 6 print statements
    print(first_name, end=" ")
    print(last_name, end="\n")
    print(address, end="\n")
    print(city, end=", ")
    print(state.upper(), end="  ")
    print(zip_code)

if __name__ == '__main__':
  
    p1(input)
    print("\n")
    p2()
    print("\n")
    p3()
